components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -39,6 +39,25 @@ pub fn concat(args: &[&Option<Bytes>]) -> Result<Option<Bytes>> {
     Ok(Some(output))
 }
 
+#[rpn_fn(varg, min_args = 2)]
+#[inline]
+pub fn concat_ws(args: &[&Option<Bytes>]) -> Result<Option<Bytes>> {
+    if let (Some(sep), Some(s1)) = (args[0], args[1]) {
+        let mut output = s1.to_vec();
+        for s in &args[2..] {
+            if let Some(s) = s {
+                output.extend_from_slice(sep);
+                output.extend_from_slice(s);
+            } else {
+                return Ok(None);
```

Comment:
This behaviour might not be correct.. According to MySQL when there are nulls the result is not null, which is very interesting.

```
mysql> select concat_ws(',', 'a', null, 'c');
--------------
select concat_ws(',', 'a', null, 'c')
--------------

+--------------------------------+
| concat_ws(',', 'a', null, 'c') |
+--------------------------------+
| a,c                            |
+--------------------------------+
1 row in set (0.04 sec)
```

Reply:
Copy that :)

Reply:
https://github.com/tikv/tikv/pull/5999 Create an PR here to fix this bug.

Reply:
Can we testing and merging it after #5999 is merged?